src/main.rs

```diff
@@ -237,22 +235,15 @@ fn show_paths(custom_pages_dir: Option<impl AsRef<Path>>) {
                 .unwrap_or_else(|_| String::from("[Invalid]"))
         },
     );
-    let custom_pages_dir = if let Some(dir) = custom_pages_dir {
-        dir.as_ref()
-            .to_path_buf()
-            .into_os_string()
-            .into_string()
-            .unwrap_or_else(|_| String::from("[Invalid]"))
-    } else {
-        // Default custom page directory is guarenteed to be Some(..), so first unwrap here is safe
-        // unless the default is changed
-        DirectoriesConfig::default()
-            .custom_pages_dir
-            .unwrap()
-            .into_os_string()
-            .into_string()
-            .unwrap_or_else(|_| String::from("[Invalid]"))
-    };
+
+    // This first unwrap will never panic because RawDirectoriesConfig implements Default. The only
+    // reason this is Option is if get_app_root() fails.
+    let custom_pages_dir = custom_pages_dir
+        .unwrap()
+        .as_ref()
+        .to_str()
+        .map(ToString::to_string)
+        .unwrap_or_else(|| String::from("[Invalid]"));
```

Comment:
Although I think I may have proposed this some time ago, you should be able to just use a `&str` here. I don't think the `unwrap` is proper here, if this variable would really always be `Some`, the argument shouldn't be an option. Some analysis is needed here

---

I am a bit confused by this function again (also outside of your diff). It feels a bit like a lot of similar pieces are coming together here, but all the information is gathered in different ways. I am not up to speed on how config / argument / env var merging is handled in tealdeer, but I feel like this should really be a method on some big instance of `Configuration` (like a singleton, but not global?) or similar that knows all the final settings. This has nothing to do with your diff though, I'm just spiraling :laughing:

Reply:
It was because I had to use get_app_root in the Default for RawDirectoriesConfig and I had I discussion with @dbrgn about whether or not the RawDirectoriesConfig::default() should panic if we unwrapped get_app_root. 

I'll try to find that discussion

Reply:
It appears that I had made the decision to use option, I think I'm going to go back and change it now that I realize how unintuitive this is.

Reply:
Well, yes. But the `show_paths` function requires the option to be `Some`, so it could just take the value. The caller would then have the `unwrap` or even better, an `expect`. Am I missing something?

Reply:
No worries :+1: